src/lib/redux/siteData.ts

```diff
@@ -35,3 +38,42 @@ export const updateTariffPlanInRedux = (tariffPlan: Partial<TariffPlanRecord>) =
 export const updateSiteInRedux = (siteUpdates : Partial<SiteRecord>) => {
   store.dispatch(updateSite(siteUpdates))
 }
+
+// Can't use "createSelector" because selectors aren't initialized before file is read
+export const selectCurrentSiteProfit = (state: RootState) => {
+  return selectCurrentPeriodTotalAmountCollected(state) - selectCurrentPeriodPurchaseRequestsApprovedTotalAmountSpent(state);
+}
+
+// Can't use "createSelector" because selectors aren't initialized before file is read
+export const selectCurrentPeriodFinancialSummary  = () => {
+  const state = store.getState();
+  const customers = selectAllCustomersArray(state);
+  const customersBilled = selectCustomersToCollect(state);
+  const customersPaid = selectCustomersDone(state);
+  const totalAmountBilled = selectCurrentPeriodTotalAmountBilled(state);
+  const totalAmountCollected =  selectCurrentPeriodTotalAmountCollected(state);
+  const totalProfit = selectCurrentSiteProfit(state);
+  const inventoryAmountApproved = selectCurrentPeriodPurchaseRequestsApprovedTotalAmountSpent(state);
+  const inventoryAmountDenied = selectCurrentPeriodPurchaseRequestsDeniedTotalAmountSpent(state);
+
+  const currentReport: FinancialSummaryRecord = {
+    id: '',
+    name: '',
+    totalCustomers: customers.length,
+    totalCustomersBilled: customersBilled.length,
+    totalCustomersPaid: customersPaid.length,
+    totalUsage: 0,
+    totalAmountBilled,
+    totalAmountCollected,
+    totalAmountSpent: 0,
+    totalProfit,
+    inventoryAmountApproved,
+    inventoryAmountDenied,
+    period: getCurrentPeriod(),
+    isapproved: false,
+    lastUpdated: '0',
```

Comment:
NIT: `lastUpdated` should be `formatDateToLocal(moment())` or whatever "right now" is. Not that we use it, but no point feeding bad data!